src/main/java/org/elasticsearch/search/highlight/HighlightPhase.java

```diff
@@ -116,9 +119,26 @@ public void hitExecute(SearchContext context, HitContext hitContext) {
 
                 Query highlightQuery = field.fieldOptions().highlightQuery() == null ? context.parsedQuery().query() : field.fieldOptions().highlightQuery();
                 HighlighterContext highlighterContext = new HighlighterContext(fieldName, field, fieldMapper, context, hitContext, highlightQuery);
-                HighlightField highlightField = highlighter.highlight(highlighterContext);
-                if (highlightField != null) {
-                    highlightFields.put(highlightField.name(), highlightField);
+
+                if (highlighter.cannotHighlight(fieldMapper) && fieldNameContainsWildcards) {
```

Comment:
When source is disabled and field is not stored highlighter.highlight() just returns null for each of the highlighters. This has not caused any problems. I changed the tests for fvh and postings below to check that search requests with '*' in the name field do not fail. That will try to highlight all meta fields. Should I add a dedicated test for fields that have no source and are not stored?